plugins/modules/ec2_vol.py

```diff
@@ -385,21 +398,35 @@ def update_volume(module, ec2_conn, volume):
 
         target_throughput = module.params.get('throughput')
         throughput_changed = False
-        if 'gp3' in [target_type, original_type]:
-            if target_throughput:
+        if target_throughput is not None:
+            if 'gp3' in [target_type, original_type]:
```

Comment:
this will collide with https://github.com/ansible-collections/amazon.aws/pull/334  
and maybe #334 should be priorised before landing this PR.